web/server/settings.py

```diff
@@ -81,10 +81,9 @@ async def get_loaded_context(settings: Settings = Depends(get_settings)) -> Cont
         )
 
 
-async def get_context(settings: Settings = Depends(get_settings)) -> Context:
+def get_context(settings: Settings = Depends(get_settings)) -> Context:
     try:
-        async with get_context_lock:
```

Comment:
I think this was added because multiple requests were asking for a context at the same time and simultaneous context creations wasn't supported. Is this not the case anymore? Also, have you checked if this works on smaller projects, e.g. the sushi example project?